app/modules/util/package.scala

```diff
@@ -0,0 +1,259 @@
+package modules
+
+import java.sql.Date
+import java.text.SimpleDateFormat
+
+import modules.util.Country
+import modules.util.Country.CountryVal
+import modules.util.Gender.GenderVal
+import modules.util.Language.LanguageVal
+import play.api.data.FormError
+import play.api.data.format.Formatter
+import play.api.mvc.QueryStringBindable
+
+import scala.language.implicitConversions
+import scala.util.{Failure, Success, Try}
+
+package object util {
+
+  def CountryFormatter: Formatter[CountryVal] = new Formatter[CountryVal] {
+    def bind(key: String, data: Map[String, String]): Either[List[FormError], CountryVal] = {
+      val value = data.getOrElse(key, "")
+      Country
+        .findByString(value)
+        .map(Right(_))
+        .getOrElse(Left(List(FormError(key, s"Country not found by value:'$value'"))))
+    }
+    def unbind(key: String, value: CountryVal): Map[String, String] = Map(key -> value.dbName)
+  }
+
+  def NationalityFormatter: Formatter[CountryVal] = new Formatter[CountryVal] {
+    def bind(key: String, data: Map[String, String]): Either[List[FormError], CountryVal] = {
+      val value = data.getOrElse(key, "")
+      Country
+        .findByString(value)
+        .map(Right(_))
+        .getOrElse(Left(List(FormError(key, s"Nationality not found by value:'$value'"))))
+    }
+    def unbind(key: String, value: CountryVal): Map[String, String] = Map(key -> value.nationality)
+  }
+
+  object Country extends Enumeration {
+    sealed case class CountryVal private[Country](dbName: String, countryName: String, nationality: String) extends super.Val
```

Comment:
Inner classes, that fulfill class instance-level functionality for Enumeration, should not leak out of Enum itself, just like in https://www.scala-lang.org/api/current/scala/Enumeration.html example, with `protected case class Val`. Instance-level functionality for Enumeration should be accessed by using conversions to inner class instance. Making inner class sealed only limits options for subclassing.